### PR TITLE
Fix 'filter features within map view' example

### PR DESCRIPTION
**Pull request by mollymerp**
This example was not doing what it is supposed to. Now it is. Flagged by @lyzidiamond from a user question.

cc @lucaswoj

### CODE REVIEW

**Comment by lucaswoj** [comment]
cc @tristen for review

**Comment by mollymerp** [comment]
_current version_
screen_shot_2016-09-19_at_2_53_09_pm

_new version_
screen_shot_2016-09-19_at_2_53_15_pm

File: docs/_posts/examples/3400-01-14-filter-features-within-map-view.html [hunk 1]
```diff
@@ -150,30 +156,33 @@
...
-        var features = map.querySourceFeatures('airports', {
-            sourceLayer: 'ne_10m_airports'
-        });
+        var features = map.queryRenderedFeatures({layers:['airport']});
 
         if (features) {
+            var renderedAbbrevs = {};
+            // It is possible for the same feature to be included twice in the output of
+            // `map.queryRenderedFeatures()` if features span tile boundaries, so we
+            // filter them out by a unique key
```
Comment (lucaswoj) [inline docs/_posts/examples/3400-01-14-filter-features-within-map-view.html]: A feature may be included more than twice in the output. Non-point features may be included an unbounded number of times in the output. 

The language in the [docs](https://www.mapbox.com/mapbox-gl-js/api/#Map#queryRenderedFeatures) is good. Feel free to borrow from there.

File: docs/_posts/examples/3400-01-14-filter-features-within-map-view.html [hunk 2]
```diff
@@ -149,31 +155,35 @@
...
+            var uniqueFeatures = features.filter(function(f) {
+                if (renderedAbbrevs[f.properties.abbrev]) {
+                    return false;
+                } else {
+                    renderedAbbrevs[f.properties.abbrev] = true;
+                    return true;
+                }
+            });
```
Comment (lucaswoj) [inline docs/_posts/examples/3400-01-14-filter-features-within-map-view.html]: This code would be more readable if you factored this code into a separate utility function

```js
function unique(array, comparator) { ... }
```

**Comment by tristen** [comment]
@mollymerp Just went to try it out when no results are found on input and `moveend` isn't repopulating results?

filters

**Comment by mollymerp** [comment]
@tristen thanks for 🎌  -- fixed in most recent commit. 

@lucaswoj changed language in comments + factored out filtering function into new util function

**Comment by tristen** [comment]
@mollymerp 👍 It's looks like you can drop these [style rules](https://github.com/mapbox/mapbox-gl-js/blob/6b958527ff2c7d81918f48d71ffa627f77e9498d/docs/_posts/examples/3400-01-14-filter-features-within-map-view.html#L26-L27) now and simplify the container structure by dropping `map-overlay-inner` and putting your elements in `map-overlay`.

**Comment by mollymerp** [comment]
I'm still seeing weird behavior though... sometimes `qRF` just doesn't seem to be returning all the features in view 😐 

screen shot 2016-09-19 at 3 27 11 pm

**Pull request merged**